kvbc/src/ror_app_state.cpp

```diff
@@ -0,0 +1,165 @@
+#include "ror_app_state.hpp"
+#include <cassert>
+#include <cstring>
+#include <unordered_map>
+#include <vector>
+#include "hash_defs.h"
+
+using namespace concordUtils;
+using namespace bftEngine::SimpleBlockchainStateTransfer;
+using namespace concord::storage;
+
+class RoRAppStateImpl : public IAppState {
+ public:
+  RoRAppStateImpl(std::shared_ptr<concord::storage::blockchain::DBKeyManipulator> keyManipulator,
+                  std::shared_ptr<concord::storage::IDBClient> localClient,
+                  std::shared_ptr<concord::storage::IDBClient> objectStoreClient)
```

Comment:
If you follow the suggestion in #406, you would pass in a std::shared_ptr<concord::storage::IObjectStoreClient> here.